infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -182,6 +193,9 @@ private void init() {
                         .setNameFormat("snapshotSyncApplyExecutor-" + session.hashCode())
                         .build());
 
+        if (session.getSubscriber().getModel().equals(LogReplication.ReplicationModel.ROUTING_QUEUES)) {
```

Comment:
Adding a comment here to explain the reason behind opening the queue will be helpful